src/types.rs

```diff
@@ -285,3 +301,27 @@ impl AssetId {
     /// The base native asset of the Fuel protocol.
     pub const BASE: AssetId = AssetId::zeroed();
 }
+
+impl UtxoId {
+    pub const fn tx_id(&self) -> TxId {
+        let r = self.0;
+
+        #[rustfmt::skip]
+        let tx_id: [u8; TxId::LEN] = [
+            r[0], r[1], r[2], r[3], r[4], r[5], r[6], r[7],
+            r[8], r[9], r[10], r[11], r[12], r[13], r[14], r[15],
+            r[16], r[17], r[18], r[19], r[20], r[21], r[22], r[23],
+            r[24], r[25], r[26], r[27], r[28], r[29], r[30], r[31],
+        ];
+
+        TxId::new(tx_id)
+    }
+
+    pub const fn output_index(&self) -> u8 {
+        self.0[TxId::LEN]
```

Comment:
This whole behavior looks like some magic. Instead of having hardcoded positions in the array, maybe we want to create a field with the output index?

Reply:
The purpose of this change is to represent the `UtxoId` as an array of 33 bytes=) Because on the `fuel-core` level, we work with arrays and slices. 

It can be a struct with two fields, but then in the `AsRef<[u8]>` implementation, we need to use `unsafe{ mem::transmute }` to convert `&UtxoId` into `&[u8; 33]` or `&[u8]`.

Reply:
I don't think it's a good approach to let serialization concerns affect how our core domain types are structured. It's a bit like the tail wagging the dog tbh.

On the other hand, copying these utxoIds every time we go to storage isn't great either.

Maybe we should consider using a vetted zerocopy lib instead of implementing transmute ourselves? e.g. https://docs.rs/zerocopy/latest/zerocopy/trait.AsBytes.html

Reply:
Did an alter implementation https://github.com/FuelLabs/fuel-types/pull/52

For me, the approach with `[u8; 33]` looks cleaner=)